.github/workflows/ci.yml

```diff
@@ -0,0 +1,69 @@
+name: C++ CI Workflow
+
+on:
+  push:
+    branches:
+      - master
```

Comment:
Is there any reason to limit this workflow to run only on master? What about new branches upstream / new branches on a fork? I'm not sure if the pipeline is triggered anyways in this case.

Reply:
This are good points, but my rationale is the following: robotology-superbuilds jobs will be quite heavy, and when you create a new commit you are typically working on docs or CMake structure, and so I think it make sense to defer running the actual CI to when a new PR is opened. We can always revise this in the future.

Reply:
Ok, good to know :+1: